libraries/chain/include/eosio/chain/apply_context.hpp

```diff
@@ -553,6 +585,9 @@ class apply_context {
 
       action_name get_sender() const;
 
+      uint32_t get_action_id() const;
+      void increment_action_id();
```

Comment:
Put the implementation of these two simple functions here so they can be inlined.

Reply:
The `trx_context` type is incomplete in the header, and I would need to also forward declare the `action_id` class to fix it.

I think it's not worth it, what do you think?

Reply:
Ah right. Fair point. Then it is fine as is.